internal/entrypoint/run.go

```diff
@@ -135,22 +137,21 @@ func Run(ctx context.Context, config *Config, exporter otlexporters.Otlexporter,
 
 			logger.WithField("number of PowerFlexClient", len(config.PowerFlexClient)).Debug("PowerFlexClient")
 
-			for key, client := range config.PowerFlexClient {
+			for key, sioConfig := range config.PowerFlexClient {
 				logger.WithField("storage system id", key).Debug("storage system id")
-				sioConfig, ok := config.PowerFlexConfig[key]
-				if !ok {
-					logger.WithError(err).WithField("storage_system_id", key).Error("no configuration found for storage_system_id")
-					continue
-				}
-				_, err := client.Authenticate(&sioConfig)
+
+				endpoint := sioConfig.Client.GetConfigConnect().Endpoint
+
+				token, err := sioConfig.TokenGetter.GetToken(ctx)
 				if err != nil {
-					logger.WithError(err).WithField("endpoint", sioConfig.Endpoint).Error("failed to authenticate with PowerFlex. retrying on next tick...")
+					logger.WithError(err).Error("getting authentication token")
```

Comment:
What will this error message print? Will it be clear enough? The one it's replacing specifies that it failed to authenticate and that it will retry.

Reply:
The `err` will only be a context error:

https://github.com/dell/karavi-metrics-powerflex/blob/a7ba83f52ff83ab956cb8d5f079c2694dd722792/internal/service/token_getter.go#L94

GetToken will only fail if the context is cancelled, which it never should be. If there are problems authenticating to the array in the token getter, it'll be logged:

https://github.com/dell/karavi-metrics-powerflex/blob/a7ba83f52ff83ab956cb8d5f079c2694dd722792/internal/service/token_getter.go#L113